src/main/java/redis/clients/jedis/JedisShardInfo.java

```diff
@@ -91,9 +102,21 @@ public void setTimeout(int timeout) {
   public String getName() {
     return name;
   }
+  
+  public int getDb() {
+    return db;
+  }
 
   @Override
   public Jedis createResource() {
-    return new Jedis(this);
+    Jedis jedis = new Jedis(this);
```

Comment:
@nykolaslima We can change BinaryClient to add setDBIndex() so initial DB index can be set, as password did.